src/peergos/server/net/DHTHandler.java

```diff
@@ -29,28 +29,37 @@
     private final Hasher hasher;
     private final BiFunction<PublicKeyHash, Integer, Boolean> keyFilter;
     private final String apiPrefix;
+    private final boolean isPublicServer;
 
     public DHTHandler(ContentAddressedStorage dht,
                       Hasher hasher,
                       BiFunction<PublicKeyHash, Integer, Boolean> keyFilter,
-                      String apiPrefix) {
+                      String apiPrefix,
+                      boolean isPublicServer) {
```

Comment:
Other option is to have a Handler superclass. call super(isPublicServer) in constructor. Then have the super.handle() deal with checking and potentially throwing 405 before calling a new method called handleRequest that each subclass overrides.

Reply:
All of these are going to be rewritten once we move to netty. I think this is simple enough for now. Then I imagine we'd have a chain of handlers, and if the first doesn't handle it, then the second does etc. i.e. an actual web framework.